### PR TITLE
Coordiantes refactor

### LINKED ISSUE #386: [REFACTOR] Unify JoystickValues

### PR DESCRIPTION
**Pull request by manucamejo**
Closes #386

### CODE REVIEW

File: server/native/gamestate/src/game.rs [hunk 1]
```diff
@@ -363,25 +367,25 @@ impl GameState {
     }
 
     pub fn position_to_direction(position: &RelativePosition) -> Direction {
```
Comment (fkrause98) [inline server/native/gamestate/src/game.rs:369]: I would move this function to where RelativePosition was defined, we have too many functions already here.

Comment (fkrause98) [reply server/native/gamestate/src/game.rs:369]: Even more so, we could make a From<Direction> for RelativePosition trait to simply use .into() when we want to do this type transformation.

Comment (manucamejo) [reply server/native/gamestate/src/game.rs:369]: we are going to refactor this code with the new distance calculation

**Review by tvillegas98 — APPROVED** [review]
Seems to be working as it should 😄

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**